frontend/src/js-admin/components/Venue.tsx

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+import React = require('react');
+import $ = require('jquery');
+import _ = require('underscore');
+import Bootstrap = require('react-bootstrap');
+
+import SeatSelector from './SeatSelector.tsx';
+import editable = require('./editables.tsx');
+import {IVenue, ISection, ISeat} from '../../../../backend/src/venue';
+
+
+export interface IVenueProps {
+  venue_id: number;
+}
+
+export interface IVenueState {
+  originalVenue?: IVenue;
+  venue: IVenue;
+}
+
+function deepClone<T extends {}>(obj: T): T {
+  return JSON.parse(JSON.stringify(obj));
```

Comment:
Hnngggh. This is correct only to some extent - fails with e.g.`deepClone({'x': new Date()})` because there is no such thing as Date in JSON, but in JS there is.

As I'm not sure if there are any specific examples that might hurt us, don't know a good simple correct way to do this, and don't want to spend lots of time for it, this is ok if you think so even after my example.